sdk/platform/http_client/nohttp/src/az_nohttp.c

```diff
@@ -1,7 +1,8 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // SPDX-License-Identifier: MIT
 
-#include <az_http_internal.h>
+#include <az_http.h>
+#include <az_http_transport.h>
```

Comment:
We actually need http_internal - it defines the signature of `az_http_client_send_request()`, and includes the headers necessary to introduce _az_http_request and az_http_response.
It is good to include http_internal .h, because if the signature in http_internal.h gets changed, you get compilation errors in this .c file.

Reply:
`az_http_client_send_request()` was moved to `az_http_transport.h`
It is no longer an internal header.
It needs to be accessed for anyone who wants to implement a transport layer.

az_http_internal is only for creating a pipeline and creating an az_http_request.
az_http_internal is only for SDK clients.

Reply:
nit:we only need http_transport.h then.

Reply:
we need http because it defines az_http_response structure

Reply:
az_http_transport.h has to include az_http.h for exactly that reason; so it is not needed here.